lib/groups.js

```diff
@@ -104,28 +139,33 @@ module.exports.joinGroup = function (joinCode, assessmentId, userId, authnUserId
 };
 
 module.exports.createGroup = function (groupName, assessmentId, userId, authnUserId, callback) {
-  const params = {
+  let params = {
     assessment_id: assessmentId,
     user_id: userId,
     authn_user_id: authnUserId,
     group_name: groupName,
+    group_role_id: null,
   };
   //alphanumeric characters only
   let invalidGroupName = true;
   const letters = /^[0-9a-zA-Z]+$/;
   if (groupName.length <= 30 && groupName.match(letters)) {
     invalidGroupName = false;
-    sqldb.query(sql.create_group, params, function (err, _result) {
-      if (!err) {
-        callback(null, true);
-      } else {
-        sqldb.query(sql.get_config_info, params, function (err, result) {
-          if (ERR(err, callback)) return;
-          const permissions = result.rows[0];
-          const uniqueGroupName = true;
-          callback(null, false, uniqueGroupName, null, permissions);
-        });
-      }
+    sqldb.query(sql.get_creator_role, params, function (err, result) {
+      if (ERR(err, callback)) return;
+      params.group_role_id = result.rows[0].id;
+      sqldb.query(sql.create_group, params, function (err, _result) {
```

Comment:
Rather than having two separate queries, I'd recommend using a CTE inside `create_group` to assign a role to the group creator.